sqldb/sqlc/migrations/000001_invoices.up.sql

```diff
@@ -10,6 +20,14 @@ CREATE TABLE IF NOT EXISTS invoices (
     -- The preimage for the hash in this invoice. Some invoices may have this 
     -- field empty, like unsettled hodl invoices or AMP invoices.
     preimage BLOB,
+    
+    -- If settled, the index is set to the current_value+1 of the settled_index
+    -- seuqence in the invoice_sequences table. If not settled,then it is set
+    -- to NULL.
+    settled_index BIGINT,
```

Comment:
Hm, what about the idea re using something more event sourced here instead of relying on the settle index value?

In other words, we'd have a table like `invoice_events`, that would look something like: https://github.com/lightninglabs/taproot-assets/blob/v0.3.1/tapdb/sqlc/migrations/000006_addr_event.up.sql#L3

Then from this we can just add new field or even new foreign key refs for arbitrary event types. This would make it easy to add events for stuff like accepting a hodl invoice, etc.